temporalio/failure.py

```diff
@@ -0,0 +1,61 @@
+from enum import IntEnum
+from typing import Any, List, Optional
+
+import temporalio.api.enums.v1
+import temporalio.api.failure.v1
+import temporalio.converter
+
+
+class FailureError(Exception):
```

Comment:
This is an odd name. Maybe `TemporalError` instead?

Reply:
There are other errors that will occur for "Temporal", this is just for https://github.com/temporalio/api/blob/5c7111f395324103bf630d6a211279adbfb5e418/temporal/api/failure/v1/message.proto#L81. There will even be one above this that is for `ServiceError`. I haven't filled this out yet, but I will shortly and the hierarchy will become a bit clearer.

I'm also trying to avoid `Temporal` in the names to avoid stutter/typing if I can.

Reply:
I agree that this error name is odd. In the other SDKs we call it `TemporalFailure` which might not be the best name but I feel like it's better than what you suggested.

Reply:
Is there a reason you don't want to make an error class per failure type?
I think it'd be more convenient to use.

Reply:
> Is there a reason you don't want to make an error class per failure type?

Not particularly. Just figured a single error type to represent https://github.com/temporalio/api/blob/master/temporal/api/failure/v1/message.proto#L81 made sense. But can change. I am reworking errors now anyways (note they were left unimplemented in this PR).

Reply:
I have reworked this into an `exceptions` module.